### ISSUE TITLE
Unusual results with hvx

### ISSUE THREAD

**Issue by cheater** [chrisnc/hvx#10]
I have created a small [script](https://gist.github.com/cheater/92998927d034e64d27055d082b75ff2c#file-solve-hs) that uses hvx. Its objective is to find solutions such that `x1`, `x2` and `x3` are in the ratio `1:3:5`, and that their sum is as close to `1100` as possible. The script as it is seems to create good solutions:
```
> solve 50000
([("x1",(1><1)
 [ 122.24454313213822 ]),("x2",(1><1)
 [ 366.6919377060333 ]),("x3",(1><1)
 [ 611.1393322799283 ]),("lineVar",(1><1)
 [ 122.1104007305691 ])],-1099.373617125954)
```
The objective used by `subgradMinimize` is a sum of two others:
```
objective = objectiveMaxSum +~ objectiveDistFromRatio
```
However, when one of the objectives is changed from this:
```
objectiveMaxSum = (cscalar (-1)) *~ sumx1x2x3
```
to this:
```
objectiveMaxSum = norm 2 $ (cscalar (-1)) *~ sumx1x2x3
```
then the results differ wildly:
```
> solve 50000
([("x1",(1><1)
 [ -1.8939468105086253e-2 ]),("x2",(1><1)
 [ -2.2246212272865742e-2 ]),("x3",(1><1)
 [ -2.011644618281285e-2 ]),("lineVar",(1><1)
 [ 8.722016213219164e-2 ])],0.609031134450206)
```
This is more than likely error on my part, however I can't imagine what's gone wrong. It seems unlikely that this is the use of `norm 2` there. The output of `solve` seems to even disregard the constraints, such as that the variables should be non-negative (`constraintsVarsPos = (>=~ (cscalar 0)) <$> vars`) or that their sum should be (`constraint2 = sumx1x2x3 >=~ (cscalar 0)`). I am likely missing some requirement in hvx. If anything, I would like to add something to the docs that tells people to watch out for a pitfall I may have fallen into. How can I figure out the reason hvx is not giving me the results I expect?

**Comment by chrisnc** [comment]
The first thing I notice is that the results are actually close to zero (relatively speaking). This gives a little bit of a clue about what's going on. I think the problem is that by applying `norm 2` to a scalar, you are actually asking for its absolute value. So you changed "minimize -1 * (x1 + x2 + x3)" into "minimize |-1 * (x1 + x2 + x3)|", which can be simplified to "minimize (x1 + x2 + x3)". Hence the optimization pushes these close to zero. I tested myself and as I upped the iteration count they hover around zero, with smaller and smaller magnitude, and positive most of the time.

If you're actually trying to do something like: "minimize -1 * ||x||_2", i.e., "maximize ||x||_2", you'll soon discover that this is not possible because it's not a convex problem. Norms are convex functions, and can only be minimized, not maximized. Maximizing the sum works because it is affine (both convex and concave), so it can be maximized or minimized.

(Accidentally commented on your gist. Moving those here.)
